pkg/resourcekeeper/statekeep.go

```diff
@@ -57,6 +63,13 @@ func (h *resourceKeeper) StateKeep(ctx context.Context) error {
 						return errors.Wrapf(err, "failed to decode resource %s from resourcetracker", mr.ResourceKey())
 					}
 					applyCtx := multicluster.ContextWithClusterName(ctx, mr.Cluster)
+					applyOncePath := h.applyOncePolicy.FindStrategy(manifest)
+					if applyOncePath != nil {
+						manifest, err = applyOnceManifest(applyCtx, h, manifest, applyOncePath)
+						if err != nil {
+							return errors.Wrapf(err, "failed to apply once resource %s from resourcetracker %s", mr.ResourceKey(), rt.Name)
+						}
+					}
```

Comment:
let's merge these lines into a common function like:

```
manifest, err := ApplyStrategies(applyCtx, h, manifest)
```

Reply:
ok, i will modify it